pkg/openstack/loadbalancer_test.go

```diff
@@ -534,3 +535,31 @@ func Test_getListenerProtocol(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateServiceAnnotations(t *testing.T) {
+	service := &corev1.Service{
+		ObjectMeta: v1.ObjectMeta{
+			Annotations: nil,
+		},
+	}
+
+	annotations := map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+	}
+
+	lbaas := LbaasV2{}
+	lbaas.updateServiceAnnotations(service, annotations)
+
+	if len(service.ObjectMeta.Annotations) != 2 {
+		t.Errorf("Expected 2 annotations, but got %d", len(service.ObjectMeta.Annotations))
+	}
+
+	if service.ObjectMeta.Annotations["key1"] != "value1" {
+		t.Errorf("Expected annotation key1 to be value1, but got %s", service.ObjectMeta.Annotations["key1"])
+	}
+
+	if service.ObjectMeta.Annotations["key2"] != "value2" {
+		t.Errorf("Expected annotation key2 to be value2, but got %s", service.ObjectMeta.Annotations["key2"])
+	}
```

Comment:
You should be able to use assert.ElementsMatch() to save quite a few lines of code.

Reply:
okay noted !